documentation/my-homeworks/bulletin-board/src/Note.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react'
+import FaPencil from 'react-icons/lib/fa/pencil'
+import FaTrash from 'react-icons/lib/fa/trash'
+import FaFloppyO from 'react-icons/lib/fa/floppy-o'
+
+class Note extends Component {
+	constructor(props) {
+		super(props)
+		this.state = {
+			editing: false
+		}
+		this.edit = this.edit.bind(this)
+		this.remove = this.remove.bind(this)
+		this.save = this.save.bind(this)
+		this.renderForm = this.renderForm.bind(this)
+		this.renderDisplay = this.renderDisplay.bind(this)
+		this.randomBetween = this.randomBetween.bind(this)
+	}
+
+	componentWillMount() {
+		this.style = {
+			right: this.randomBetween(0, window.innerWidth - 150, 'px'),
+			top: this.randomBetween(0, window.innerHeight - 150, 'px'),
+			transform: `rotate(${this.randomBetween(-25, 25, 'deg')})`
+		}
+	}
+
+	randomBetween(x, y, s) {
+		return x + Math.ceil(Math.random() * (y-x)) + s
+	}
+
+	componentDidUpdate() {
+		var textArea
+		if(this.state.editing) {
+			textArea = this._newText
+			textArea.focus()
+			textArea.select()
+		}
+
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
+		return (
+			this.props.children !== nextProps.children || this.state !== nextState
+		)
+	}
+
+	edit() {
+		this.setState({
+			editing: true
+		})
+	}
+
+	remove() {
+		this.props.onRemove(this.props.index)
```

Comment:
If you are going to make a call `this.props.onRemove(this.props.index)` only, so there's no need to create **remove** function. Just use the `this.props.onRemove(this.props.index)` directly on JSX code.